hist/histpainter/src/THistPainter.cxx

```diff
@@ -4041,49 +4041,49 @@ Int_t THistPainter::MakeChopt(Option_t *choptin)
    l = strstr(chopt,"LEGO");
    if (l) {
       Hoption.Scat = 0;
-      Hoption.Lego = 1; strncpy(l,"    ",4);
+      Hoption.Lego = 1; strncpy(l,"    ",5);
       if (l[4] == '1') { Hoption.Lego = 11; l[4] = ' '; }
       if (l[4] == '2') { Hoption.Lego = 12; l[4] = ' '; }
       if (l[4] == '3') { Hoption.Lego = 13; l[4] = ' '; }
       if (l[4] == '4') { Hoption.Lego = 14; l[4] = ' '; }
       if (l[4] == '9') { Hoption.Lego = 19; l[4] = ' '; }
-      l = strstr(chopt,"FB"); if (l) { Hoption.FrontBox = 0; strncpy(l,"  ",2); }
-      l = strstr(chopt,"BB"); if (l) { Hoption.BackBox = 0;  strncpy(l,"  ",2); }
-      l = strstr(chopt,"0");  if (l) { Hoption.Zero = 1;  strncpy(l," ",1); }
+      l = strstr(chopt,"FB"); if (l) { Hoption.FrontBox = 0; strncpy(l,"  ",3); }
+      l = strstr(chopt,"BB"); if (l) { Hoption.BackBox = 0;  strncpy(l,"  ",3); }
+      l = strstr(chopt,"0");  if (l) { Hoption.Zero = 1;  strncpy(l," ",2); }
    }
 
    l = strstr(chopt,"SURF");
    if (l) {
       Hoption.Scat = 0;
-      Hoption.Surf = 1; strncpy(l,"    ",4);
+      Hoption.Surf = 1; strncpy(l,"    ",5);
       if (l[4] == '1') { Hoption.Surf = 11; l[4] = ' '; }
       if (l[4] == '2') { Hoption.Surf = 12; l[4] = ' '; }
       if (l[4] == '3') { Hoption.Surf = 13; l[4] = ' '; }
       if (l[4] == '4') { Hoption.Surf = 14; l[4] = ' '; }
       if (l[4] == '5') { Hoption.Surf = 15; l[4] = ' '; }
       if (l[4] == '6') { Hoption.Surf = 16; l[4] = ' '; }
       if (l[4] == '7') { Hoption.Surf = 17; l[4] = ' '; }
-      l = strstr(chopt,"FB");   if (l) { Hoption.FrontBox = 0; strncpy(l,"  ",2); }
-      l = strstr(chopt,"BB");   if (l) { Hoption.BackBox = 0;  strncpy(l,"  ",2); }
+      l = strstr(chopt,"FB");   if (l) { Hoption.FrontBox = 0; strncpy(l,"  ",3); }
```

Comment:
This will cut the string short (inserting a '\0') and likely introduce bugs.

Reply:
Yes, that's why I closed this PR when I arrived this morning. I am now looking for some better fix. May be using memcpy instead of strncpy: memcpy(l,"  ",2) ... it seems to work for me on mac (string is ok and gcc8 is happy). I am just wondering if it is something which will work everywhere.